src/query_param.c

```diff
@@ -38,27 +38,21 @@ QueryParam *NewNumericFilterQueryParam_WithParams(struct QueryParseCtx *q, Query
   return ret;
 }
 
-QueryParam *NewVectorFilterQueryParam_WithParams(struct QueryParseCtx *q, QueryToken *vec, QueryToken *type, QueryToken *value) {
+// TODO: to be more generic, consider using variadic function, or use different functions for each command
+QueryParam *NewVectorFilterQueryParam_WithParams(struct QueryParseCtx *q, VectorQueryType type, QueryToken *value, QueryToken *vec) {
```

Comment:
why this function is called this way? this is just a normal topK query, no? why do we need VectorQuerytype?

Reply:
I wanted this function to be generic and handle all types of vecsim command in the future. as I mentioned in the comment above, we can also have a function for every query type.